src/renderer/renderScene.ts

```diff
@@ -291,6 +295,10 @@ export const renderScene = (
       });
   }
 
+  visibleElements.forEach((element) => {
+    renderLinkIcon(element, context, appState);
+  });
```

Comment:
This kinda kills performance when there are lots of links on the screen. Can we render this onto the cached canvas same as we do with elements themselves?

AFAIK It'd require to do 2 things:

1. account for the increased dimensions of the canvas in the top-right corner
2. invalidate the cache on selection/deselection since we'll need to rerender with/without the icon

Reply:
If nothing else, I'd move the rendering logic into `renderElement` so that when we start implementing tiling optimizations we don't need to rewrite this part of the codebase. But I'd still like to investigate the caching optim first.

Reply:
`renderElement` I had tried earlier and that would need updating canvas dims, updating coords of resize/ rotate handlers, update element coords and more hence I updated the approach to render it outside element canvas. I will check on caching

Reply:
(By caching I meant putting it inside the element canvas in renderElement)